app/controllers/provider/sessions_controller.rb

```diff
@@ -87,12 +88,13 @@ def auth_params
   end
 
   def sso_params
-    params.permit(:token, :expires_at, :redirect_url, :system_name, :code).merge(request: request)
+    params.permit(%i[token expires_at redirect_url system_name code]).merge(request: request).to_h
```

Comment:
Why `to_h`? It is already permitted so it should not be an issue. If it is because of a warning with unsecure redirect then OK